sql/hazard-updates.sql

```diff
@@ -0,0 +1,151 @@
+--
+-- Hazard Database Schema Updates
+-- Intended to store event sets for both deterministic and probabilistic 
+-- analyses
+--
+
+--
+-- NOTE please execute commands in common.sql before executing this file
+--
+
+--
+-- Use transaction to prevent partial execution
+--
+START TRANSACTION;
+
+-- Usual settings copied from pg_dump output
+SET statement_timeout = 0;
+SET lock_timeout = 0;
+SET client_encoding = 'UTF8';
+SET standard_conforming_strings = on;
+SET check_function_bodies = false;
+SET client_min_messages = warning;
+SET row_security = off;
+SET default_with_oids = false;
+
+SET search_path = hazard, pg_catalog, public;
+
+-----------------------------------------------------------------------------
+-- Add boolean field is_prob - TRUE for probabilistic, FALSE for deterministic
+ALTER TABLE hazard.event_set ADD COLUMN is_prob BOOLEAN NOT NULL DEFAULT FALSE;
+
+--
+-- Enumerated type for hazard calculaton method
+--
+CREATE TYPE hazard.calc_method_enum AS ENUM (
+	'INF',					-- Inferred
+	'SIM',					-- Simulation
+	'OBS'					-- Observed
+);                                                                              
+COMMENT ON TYPE hazard.calc_method_enum IS 'Hazard Calculation Methods';
+
+    --  Earthquake: ground-motion               QGM                             
+    --              primary surface rupture     Q1R                             
+    --              secondary surface rupture   Q2R                             
+    --              liquefaction                QLI                             
+    --                                                                          
+    --  Flood:      water depth                 FWD                             
+    --                                                                          
+    --  Landslide:  rock fall                   LRF                             
+    --              debris flow                 LDF                             
+    --                                                                          
+    --  Storm Surge:                                                            
+    --              inundation                  SIN                             
+    --                                                                          
+    --  Volcaninc Eruption:                                                     
+    --              ash fall                    VAF                             
+
+
+--
+-- Enumerated type for hazard process type
+--
+CREATE TYPE hazard.process_type_enum AS ENUM (
+	'DRT', 					-- Drought
```

Comment:
I see why you have problems with ENUM. It is a type that should be use for enumerations that never change (i.e. the 7 days of the week) while the hazard kinds will likely increase in the future forcing you to update the type.

Reply:
Yes, for  both hazard and process type I have reverted to lookup tables.